File: patron/src/main/java/com/prolificinteractive/patrons/IntPreference.java
package com.prolificinteractive.patrons;

import android.content.SharedPreferences;

/**
 * Preference wrapper for {@link Integer}.
 */
public class IntPreference extends BasePreference<Integer> {
  public IntPreference(final SharedPreferences preferences, final String key) {
    this(preferences, key, 0);
  }

  public IntPreference(
      final SharedPreferences preferences,
      final String key,
      final int defaultValue) {
    super(preferences, key, defaultValue);
  }

  @Override public Integer get() {
    return preferences.getInt(key, defaultValue);
  }

  @Override public void set(final Integer value) {
    preferences.edit().putInt(key, value).apply();
  }
}

File: patron/src/main/java/com/prolificinteractive/patrons/IntPreference.kt
package com.prolificinteractive.patrons

import android.content.SharedPreferences

/**
 * Preference wrapper for [Integer].
 */
class IntPreference @JvmOverloads constructor(
    preferences: SharedPreferences,
    key: String,
    defaultValue: Int? = null
) : BasePreference<Int?>(preferences, key, defaultValue) {

  override fun get() = preferences.getInt(key, defaultValue ?: 0)

  override fun set(value: Int?) = when (value) {
    null -> delete()
    else -> preferences.edit().putInt(key, value).apply()
  }
}
